worlds/AutoWorld.py

```diff
@@ -476,17 +476,15 @@ def create_group(cls, multiworld: "MultiWorld", new_player_id: int, players: Set
         return group
 
     # decent place to implement progressive items, in most cases can stay as-is
-    def collect_item(self, state: "CollectionState", item: "Item", remove: bool = False) -> Optional[str]:
+    def collect_item(self, state: "CollectionState", item: "Item", remove: bool = False) -> str:
         """
-        Collect an item name into state. For speed reasons items that aren't logically useful get skipped.
+        Collect an item name into state.
         Collect None to skip item.
```

Comment:
this isn't true anymore. since we have to change this docstring anyways, is should also be updated to include the comment above it. maybe a better version of "get the item name to be collected into state for a particular item. decent place to implement progressive items, i.e. convert "Progressive Sword" into "Fighter's Sword""

Reply:
good call